src/lib/services/HttpProxy/HttpProxy.ts

```diff
@@ -51,6 +51,20 @@ export function useHttpProxy(): IHttpProxy {
 						const isFresh = now < expiry;
 
 						if (online === null || isFresh) {
+							if (isBinaryRequest && cachedData?.__binary) {
+								// RECONSTRUCT a fresh Blob URL from the stored bytes and type
+								const bytes = cachedData.bytes instanceof ArrayBuffer
+									? new Uint8Array(cachedData.bytes)
+									: Uint8Array.from(cachedData.bytes);
+								const blob = new Blob([bytes], { type: cachedData.contentType || 'application/octet-stream' });
```

Comment:
`util.toU8` does exactly this (and more) :slightly_smiling_face:
```suggestion
								const blob = new Blob([toU8(cachedData.bytes)], { type: cachedData.contentType || 'application/octet-stream' });
```